flake.nix

```diff
@@ -0,0 +1,26 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    haskell-flake.url = "github:srid/haskell-flake";
+  };
+  outputs = inputs@{ self, nixpkgs, flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = nixpkgs.lib.systems.flakeExposed;
+      imports = [
+        inputs.haskell-flake.flakeModule
+      ];
+      perSystem = { self', pkgs, ... }: {
+        haskellProjects.default = { };
+      };
+      flake.haskellFlakeProjectModules.output = { pkgs, ... }: {
+        source-overrides = {
+          prometheus-client = ./prometheus-client;
+          prometheus-proc = ./prometheus-proc;
+          prometheus-metrics-ghc = ./prometheus-metrics-ghc;
+          wai-middleware-prometheus = ./wai-middleware-prometheus;
+          prometheus-haskell-example = ./example;
+        };
+      };
```

Comment:
It should be possible to create a `flake-parts` module option (either as part of `haskell-flake` or outside of it) to auto-generate this. But for now, manually writing it is good until we get a good intuition of how the various projects compose and use them correctly.

Reply:
Just generalized this. 

I'll leave this PR as draft, while I figure out where to put this entire generalized `flake.haskellFlakeProjectModules` thing so it can be reused in other repos.